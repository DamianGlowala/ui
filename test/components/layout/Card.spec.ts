import { describe, it, expect } from 'vitest'
import { UCard } from '#components'
import type { TypeOf } from 'zod'
import ComponentRender from '../component-render'

describe('Card', () => {
  it.each([
    ['basic case', { props: { ui: { body: { base: '' } } } }],
    ['with as', { props: { as: 'li', ui: { body: { base: '' } } } }],
    ['with default slot', { slots: { default: 'default slot' }, props: { ui: { body: { base: '' } } } }],
    ['with header slot', { slots: { header: 'Header slot' }, props: { ui: { body: { base: '' }, header: { base: '' } } } }],
    ['with footer slot', { slots: { footer: 'Footer slot' }, props: { ui: { body: { base: '' }, footer: { base: '' } } } }],
    // @ts-ignore
  ])('renders %s correctly', async (nameOrHtml: string, options: TypeOf<typeof Card.props>) => {
    const html = await ComponentRender(nameOrHtml, options, UCard)
    expect(html).toMatchSnapshot()
  })
})
