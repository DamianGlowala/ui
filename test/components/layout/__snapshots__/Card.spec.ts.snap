// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Card > renders basic case correctly 1`] = `
"<div class="rounded-lg divide-y divide-gray-200 dark:divide-gray-800 ring-1 ring-gray-200 dark:ring-gray-800 shadow bg-white dark:bg-gray-900">
  <!--v-if-->
  <!--v-if-->
  <!--v-if-->
</div>"
`;

exports[`Card > renders with as correctly 1`] = `
"<li class="rounded-lg divide-y divide-gray-200 dark:divide-gray-800 ring-1 ring-gray-200 dark:ring-gray-800 shadow bg-white dark:bg-gray-900">
  <!--v-if-->
  <!--v-if-->
  <!--v-if-->
</li>"
`;

exports[`Card > renders with default slot correctly 1`] = `
"<div class="rounded-lg divide-y divide-gray-200 dark:divide-gray-800 ring-1 ring-gray-200 dark:ring-gray-800 shadow bg-white dark:bg-gray-900">
  <!--v-if-->
  <div class="">default slot</div>
  <!--v-if-->
</div>"
`;

exports[`Card > renders with footer slot correctly 1`] = `
"<div class="rounded-lg divide-y divide-gray-200 dark:divide-gray-800 ring-1 ring-gray-200 dark:ring-gray-800 shadow bg-white dark:bg-gray-900">
  <!--v-if-->
  <!--v-if-->
  <div class="">Footer slot</div>
</div>"
`;

exports[`Card > renders with header slot correctly 1`] = `
"<div class="rounded-lg divide-y divide-gray-200 dark:divide-gray-800 ring-1 ring-gray-200 dark:ring-gray-800 shadow bg-white dark:bg-gray-900">
  <div class="">Header slot</div>
  <!--v-if-->
  <!--v-if-->
</div>"
`;
